### PR TITLE
feat: customize ui for array, object and field error templates

### PR DESCRIPTION
**Pull request by eshankvaish**
# Description

- Customize UI for array and map/object fields
- Add templates for the Add & Remove button
- Update the field error template

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/common/RJSF/templates/ArrayFieldTemplate.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    const uiOptions = getUiOptions(uiSchema)
+    const ArrayFieldItemTemplate = getTemplate<'ArrayFieldItemTemplate'>('ArrayFieldItemTemplate', registry, uiOptions)
+    const label = uiOptions.title || title
+
+    return (
+        <fieldset className={className} id={idSchema.$id}>
+            {/* Show the label here in case there are no items, otherwise handled by Field Template */}
+            {items.length ? (
+                <>
+                    {items?.map(({ key, ...itemProps }: ArrayFieldTemplateItemType, index) => {
```
Comment (vivek-devtron) [inline src/components/common/RJSF/templates/ArrayFieldTemplate.tsx:29]: we can rmeove this optional check as it is already getting checked earlier

File: src/components/common/RJSF/templates/ObjectFieldTemplate.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                        shouldAlignCenter={false}
+                    >
+                        <div>{Properties}</div>
+                    </FieldRowWithLabel>
+                    {ActionButton}
+                </>
+            )
+        } else {
+            return (
+                <>
```
Comment (vivek-devtron) [inline src/components/common/RJSF/templates/ObjectFieldTemplate.tsx:62]: We can remove this <></>

File: src/components/common/RJSF/templates/ArrayFieldTemplate.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    return (
+        canAdd && (
+            <AddButton onClick={onAddClick} disabled={disabled || readonly} uiSchema={uiSchema} registry={registry} />
+        )
+    )
+}
+
+export const ArrayFieldTemplate = (props: ArrayFieldTemplateProps) => {
+    const { canAdd, className, disabled, idSchema, uiSchema, items, onAddClick, readonly, registry, required, title } =
```
Comment (vivek-devtron) [inline src/components/common/RJSF/templates/ArrayFieldTemplate.tsx:18]: Please destructure the props in same line as the component

File: src/components/common/RJSF/templates/ObjectFieldTemplate.tsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    ObjectFieldTemplateProps,
+    canExpand,
+    getTemplate,
+    getUiOptions,
+    titleId,
+} from '@rjsf/utils'
+import { FieldRowWithLabel } from '../common/FieldRow'
+
+const Field = (props: ObjectFieldTemplateProps) => {
+    const {
```
Comment (vivek-devtron) [inline src/components/common/RJSF/templates/ObjectFieldTemplate.tsx:13]: same here

File: src/components/common/RJSF/templates/ObjectFieldTemplate.tsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    return (
+        <>
+            {Properties}
+            {ActionButton}
+        </>
+    )
+}
+
+export const ObjectFieldTemplate = (props: ObjectFieldTemplateProps) => {
+    const { idSchema, registry, required, schema, title, uiSchema } = props
```
Comment (vivek-devtron) [inline src/components/common/RJSF/templates/ObjectFieldTemplate.tsx:76]: same here

Comment (eshankvaish) [reply src/components/common/RJSF/templates/ObjectFieldTemplate.tsx:76]: props required here for passing down to the Field component

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**